### PR TITLE
Fix bug in toComplexWithDefault

### PR DESCRIPTION
**Pull request by zasdfgbnm**
I don't think this method is used anywhere, so I don't know how to test it. But the diff should justify itself.

### CODE REVIEW

File: torch/csrc/utils/python_arg_parser.h [hunk 1]
```diff
@@ -574,7 +574,7 @@ inline c10::complex<double> PythonArgs::toComplex(int i) {
 
 inline c10::complex<double> PythonArgs::toComplexWithDefault(int i, c10::complex<double> default_value) {
   if (!args[i]) return default_value;
-  return toDouble(i);
+  return toComplex(i);
```
Comment (anjali411) [inline torch/csrc/utils/python_arg_parser.h:577]: hmm not sure where this function is being used. we should add `toComplex` to this [dict](https://github.com/pytorch/pytorch/blob/a860be898ee0fbafac14bf68e920187a76ded70d/tools/autograd/gen_python_functions.py#L268) in `gen_python_functions.py`. After addition to the dict, I think this should work:
```
>>> torch.tensor(2).complex()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Tensor' object has no attribute 'complex'
``` 
similar to

```
>>> torch.tensor(2).double()
tensor(2., dtype=torch.float64)
>>> torch.tensor(2).bool()
tensor(True)
```
For `toComplex`, we can add a test here: https://github.com/pytorch/pytorch/blob/a860be898ee0fbafac14bf68e920187a76ded70d/aten/src/ATen/test/scalar_test.cpp#L55.

Comment (zasdfgbnm) [reply torch/csrc/utils/python_arg_parser.h:577]: Tested that adding this does not help with `torch.tensor(2).complex()`. I think this part of code is only used to convert python values to C++ values? For example if you have an operator that has
```
myop(Tensor self, complex?value)
```
Then this `toComplexWithDefault` will be used. But now I don't think we have such operator.

**Review by anjali411 — APPROVED** [review]

**Comment by anjali411** [comment]
@zasdfgbnm can you rebase and sign the CLA if you haven't already?

**Comment by zasdfgbnm** [comment]
@anjali411 My CLA is resolved

**Pull request closed without merging**